deploy/examples/create-external-cluster-resources.py

```diff
@@ -693,6 +699,13 @@ def get_ceph_external_mon_data(self):
                 "Only 'v2' address type is enabled, user should also enable 'v1' type as well"
             )
         ip_port = str(q_leader_details["public_addr"].split("/")[0])
+
+        if self._arg_parser.v2_port_enable:
+            if q_leader_addrvec[0]["type"] == "v2":
```

Comment:
the top condition should the length ,

Currently it will work totally fine, but just for the clarity I think we should check the length first

And also if the user has pass `self._arg_parser.v2_port_enable `  and the length is not greater than one we can stderr the error

Reply:
> the top condition should the length ,
Currently it will work totally fine, but just for the clarity I think we should check the length first

What benefit do we ge on doing that? Anyway, IIUC the probablity of `v2` being in `1nth` position, is very less. So, why check that later?

> And also if the user has pass self._arg_parser.v2_port_enable  and the length is not greater than one we can stderr the error

Sure, will update that.

Reply:
Technically, if we (almost) always confident that the latter will be the case then I think we should think from the perf angle rather than code readability here.

Reply:
Ack, understood the issue. Updating the code, to make use of best of both scenarios.